CMakeLists.txt

```diff
@@ -125,6 +125,7 @@ SET(SOURCES
   include/libfreenect2/packet_pipeline.h
   include/internal/libfreenect2/packet_processor.h
   include/libfreenect2/registration.h
+  include/libfreenect2/cuda_registration.h
```

Comment:
The API specific implementation class should be put in `registration.h`, not `cuda_registration.h`. See how CUDA depth processor doesn't have its own header instead its definition is in `depth_packet_processor.h`.

Here was the API design I outlined https://github.com/OpenKinect/libfreenect2/issues/744#issuecomment-255842615